containers/src/Kokkos_DynRankView.hpp

```diff
@@ -434,99 +489,18 @@ class DynRankView : public ViewTraits<DataType, Properties...> {
   // Domain rank and extents
 
   //  enum { Rank = map_type::Rank }; //Will be dyn rank of 7 always, keep the
   //  enum?
 
-  template <typename iType>
-  KOKKOS_INLINE_FUNCTION constexpr std::enable_if_t<std::is_integral_v<iType>,
-                                                    size_t>
-  extent(const iType& r) const {
-    return m_map.extent(r);
-  }
-
-  template <typename iType>
-  KOKKOS_INLINE_FUNCTION constexpr std::enable_if_t<std::is_integral_v<iType>,
-                                                    int>
-  extent_int(const iType& r) const {
-    return static_cast<int>(m_map.extent(r));
-  }
-
-  KOKKOS_INLINE_FUNCTION constexpr typename traits::array_layout layout() const;
-
   //----------------------------------------
   /*  Deprecate all 'dimension' functions in favor of
    *  ISO/C++ vocabulary 'extent'.
    */
 
-  KOKKOS_INLINE_FUNCTION constexpr size_t size() const {
-    return m_map.extent(0) * m_map.extent(1) * m_map.extent(2) *
-           m_map.extent(3) * m_map.extent(4) * m_map.extent(5) *
-           m_map.extent(6) * m_map.extent(7);
-  }
-
-  KOKKOS_INLINE_FUNCTION constexpr size_t stride_0() const {
-    return m_map.stride_0();
-  }
-  KOKKOS_INLINE_FUNCTION constexpr size_t stride_1() const {
-    return m_map.stride_1();
-  }
-  KOKKOS_INLINE_FUNCTION constexpr size_t stride_2() const {
-    return m_map.stride_2();
-  }
-  KOKKOS_INLINE_FUNCTION constexpr size_t stride_3() const {
-    return m_map.stride_3();
-  }
-  KOKKOS_INLINE_FUNCTION constexpr size_t stride_4() const {
-    return m_map.stride_4();
-  }
-  KOKKOS_INLINE_FUNCTION constexpr size_t stride_5() const {
-    return m_map.stride_5();
-  }
-  KOKKOS_INLINE_FUNCTION constexpr size_t stride_6() const {
-    return m_map.stride_6();
-  }
-  KOKKOS_INLINE_FUNCTION constexpr size_t stride_7() const {
-    return m_map.stride_7();
-  }
-
-  template <typename iType>
-  KOKKOS_INLINE_FUNCTION void stride(iType* const s) const {
-    m_map.stride(s);
-  }
-
-  //----------------------------------------
-  // Range span is the span which contains all members.
-
-  using reference_type = typename map_type::reference_type;
-  using pointer_type   = typename map_type::pointer_type;
-
-  enum {
-    reference_type_is_lvalue_reference =
-        std::is_lvalue_reference_v<reference_type>
-  };
-
-  KOKKOS_INLINE_FUNCTION constexpr size_t span() const { return m_map.span(); }
-  KOKKOS_INLINE_FUNCTION constexpr bool span_is_contiguous() const {
-    return m_map.span_is_contiguous();
-  }
-  KOKKOS_INLINE_FUNCTION constexpr pointer_type data() const {
-    return m_map.data();
-  }
-  KOKKOS_INLINE_FUNCTION constexpr bool is_allocated() const {
-    return (m_map.data() != nullptr);
-  }
-
-  //----------------------------------------
-  // Allow specializations to query their specialized map
-  KOKKOS_INLINE_FUNCTION
-  const Kokkos::Impl::ViewMapping<traits, typename traits::specialize>&
-  impl_map() const {
-    return m_map;
-  }
```

Comment:
Is Sacado/Stokhos using this function?
```
$ git grep -n "impl_map" packages/sacado/ packages/stokhos/
packages/sacado/src/KokkosExp_View_Fad.hpp:180:  return view.impl_map().dimension_scalar();
packages/sacado/src/Kokkos_DynRankView_Fad.hpp:988:  return view.impl_map().dimension_scalar();
packages/sacado/src/Kokkos_DynRankView_Fad.hpp:1261:  layout.stride[src.rank()] = src.impl_map().stride_scalar();
packages/sacado/src/Kokkos_DynRankView_Fad.hpp:1306:  auto layout = v.impl_map().layout();
packages/stokhos/src/sacado/kokkos/pce/KokkosExp_View_UQ_PCE_Contiguous.hpp:112:  return view.impl_map().dimension_scalar();
packages/stokhos/src/sacado/kokkos/pce/KokkosExp_View_UQ_PCE_Contiguous.hpp:121:  return view.impl_map().cijk();
packages/stokhos/src/sacado/kokkos/pce/KokkosExp_View_UQ_PCE_Contiguous.hpp:129:  return view.impl_map().is_allocation_contiguous();
packages/stokhos/src/sacado/kokkos/pce/KokkosExp_View_UQ_PCE_Contiguous.hpp:218:                             src.impl_map().cijk()), layout);
packages/stokhos/src/sacado/kokkos/pce/KokkosExp_View_UQ_PCE_Contiguous.hpp:257:                             src.impl_map().cijk()), layout);
packages/stokhos/src/sacado/kokkos/pce/KokkosExp_View_UQ_PCE_Contiguous.hpp:271:    view_alloc(src.label(), src.impl_map().cijk()),layout);
packages/stokhos/src/sacado/kokkos/pce/KokkosExp_View_UQ_PCE_Contiguous.hpp:292:                             src.impl_map().cijk()), layout);
packages/stokhos/src/sacado/kokkos/pce/KokkosExp_View_UQ_PCE_Contiguous.hpp:332:                             src.impl_map().cijk()), layout);
packages/stokhos/src/sacado/kokkos/pce/KokkosExp_View_UQ_PCE_Contiguous.hpp:347:    view_alloc(src.label(), wi, src.impl_map().cijk()), layout);
packages/stokhos/src/sacado/kokkos/pce/KokkosExp_View_UQ_PCE_Contiguous.hpp:382:      view_alloc(WithoutInitializing, label, src.impl_map().cijk()), layout};
packages/stokhos/src/sacado/kokkos/vector/KokkosExp_View_MP_Vector_Contiguous.hpp:103:  return view.impl_map().dimension_scalar();
```
or are they only using `View` and not `DynRankView`?

Reply:
They are likely using this function (in the DynRankView_Fad thing), but we broke them already :-(. I will do a post cleanup where I will fix them after the fact in one fell swoop.